fix(migrations): combine newly-added imports in import manager

crisbeto opened the pull request:
Fixes that imports weren't being combined in the `ImportManager` when multiple new imports are added for the same file. This wasn't a problem for previous schematics that used the manager, but it'll come up in some of the new ones.

devversion commented on packages/core/schematics/utils/import_manager.ts:
```diff
@@ -200,6 +209,32 @@ export class ImportManager {
       recorder.updateExistingImport(namedBindings, newNamedBindingsText);
     });
+
+    this.newImports.forEach(({importStartIndex, defaultImports, namedImports}, sourceFile) => {
```
Nice, I'm glad it's already doing the "first record everything" and then "apply" model, so that this change could be made easily 👍

devversion commented on packages/core/schematics/utils/import_manager.ts:
```diff
@@ -29,6 +29,13 @@ export class ImportManager {
+  /** Map of source files and the new imports that have to be added to them. */
+  private newImports: Map<ts.SourceFile, {
+    importStartIndex: number,
+    defaultImports: Map<string, ts.Identifier[]>,
```
Do you recall in what cases the default import for a module needs multiple identifiers? Aren't we already deduping/using the cached identifiers. Maybe should be just a single `ts.Identifier`?

```suggestion
    defaultImports: Map<string, ts.Identifier>,
```

crisbeto replied:
I don't remember exactly why it was written this way and we don't have any using it so it's hard to tell so I kept the default import logic as it was. My best guess is that it might not be safe to reuse a default import to a file, because it can be referenced under different names in the same file.

devversion replied:
I think we would never hit the case where multiple identifiers would be pushed to the array:

see: https://github.com/angular/angular/blob/f024ce44512ce92416f7af7a7008b02a2bdcd6cc/packages/core/schematics/utils/import_manager.ts#L67

crisbeto replied:
I see. Changed.

devversion reviewed (approved):
Nice! Hopefully the import manager can soon be shared across all of our schematics/migrations

crisbeto committed:
fix(migrations): combine newly-added imports in import manager

Fixes that imports weren't being combined in the `ImportManager` when multiple new imports are added for the same file. This wasn't a problem for previous schematics that used the manager, but it'll come up in some of the new ones.

Also moves the logic for writing new imports into `recordChanges`, instead of `addImportToSourceFile`.

devversion commented:
This PR was merged into the repository by commit cc284afbbc33b91884882204c5958a44a5d11392.
